### PR TITLE
Modular pkg and tests

### PR DESCRIPTION
**Pull request by leonbotros**
This PR adds modularity wrt authentication method to the PKG. It is implemented as follows:

- The authentication middleware wraps the key service and it responsible for 
1. authenticating a user,
2. extracting (validated) credentials/attributes,
3. deriving an identity (`<K as IBKEM>::Id`) for a scheme.
The middleware adds the identity to the request-local extensions. The key service extracts the key and adds it the the response-local extensions after which the middleware is responsible for creating the response. This way a key cannot end up in a response "by accident".

This PR also includes testing middleware that receives an IRMA policy and performs no authentication. This way, some PKG endpoints can be tested without testing the IRMA part of the process. Some basic tests and a full round of encapsulation using the keys obtained via the API are included.

This PR does make breaking changes to the API, specifically, the endpoints that are used for IRMA authentication are now prefixed with ```/irma```.

### CODE REVIEW

**Review by Iso5786 — COMMENTED** [review]
Also, update documentation

File: irmaseal-pkg/src/middleware/irma.rs [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+#[derive(Clone)]
+enum Auth {
+    // Check the ongoing session at {irma_url}/session/{token} associated with this token.
+    Token(String),
+
+    // Retrieve JWT decoding key from irma_url and retrieve the session result by checking the JWT in the request.
+    Jwt(DecodingKey),
+}
+
+#[doc(hidden)]
```
[the remark this answers is not in the document]
Comment (leonbotros) [reply irmaseal-pkg/src/middleware/irma.rs:53]: It's a binary crate currently not published on crates.io.

File: irmaseal-pkg/src/middleware/irma.rs [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+                        })?;
+
+                    // It is not allowed to ask for USKs with a timestamp beyond the expiry date.
+                    if timestamp > decoded.claims.exp {
+                        return Err(crate::Error::ChronologyError.into());
+                    }
+
+                    SessionResult {
+                        token: decoded.claims.token,
+                        sessiontype: decoded.claims.r#type,
```
[the remark this answers is not in the document]
Comment (leonbotros) [reply irmaseal-pkg/src/middleware/irma.rs:134]: This is a struct defined in https://crates.io/crates/irma, but otherwise I agree

**Comment by leonbotros** [comment]
Checks do not all succeed because I changed the them to a matrix strategy, but they are the same tests.
PKG API should now also be fully backwards compatible with older clients.

**Pull request merged**